scenarios/kubernetes_bazel.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs bazel build/test for current repo."""
+
+import argparse
+import os
+import subprocess
+import sys
+
+ORIG_CWD = os.getcwd()
+
+def test_infra(*paths):
+    """Return path relative to root of test-infra repo."""
+    return os.path.join(ORIG_CWD, os.path.dirname(__file__), '..', *paths)
+
+def check(*cmd):
+    """Log and run the command, raising on errors."""
+    print >>sys.stderr, 'Run:', cmd
+    subprocess.check_call(cmd)
+
+
+def echo_result(res):
+    """echo error message bazed on value of res"""
+    echo_map = {
+        0:'Success',
+        1:'Build failed',
+        2:'Bad environment or flags',
+        3:'Build passed, tests failed or timed out',
+        4:'Build passed, no tests found',
+        5:'Interrupted'
+    }
+    print echo_map.get(res, 'Unknown exit code : %s' % res)
+
+
+BUILD_MAP = {
+    'k8s' : ['make', 'bazel-build'],
+    'test-infra' : ['bazel', 'build', '//...']
+}
+
+RELEASE_MAP = {
+    'k8s' : ['make', 'bazel-release']
+}
+
+TEST_MAP = {
+    'k8s' : ['make', 'bazel-test'],
+    'test-infra' : [
+        'verify/bazel-test-affected.sh',
+        '%d...%d' % (os.environ['PULL_BASE_SHA'], os.environ['PULL_PULL_SHA'])
+        ]
+}
+
+
+def main(mode, build, release, test, gcs):
+    """Trigger a bazel build/test run, and upload results."""
+    check('bazel', 'clean', '--expunge')
+    res = 0
+    try:
+        if build:
+            check(*BUILD_MAP[mode])
+        if release:
+            check(*RELEASE_MAP[mode])
+        if test:
+            check(*TEST_MAP[mode])
+    except subprocess.CalledProcessError as exp:
+        res = exp.returncode
+
+    if release and mode == 'k8s' and res == 0:
+        with open('bazel-genfiles/version') as fp:
+            version = fp.read()
+        if not version:
+            print 'Kubernetes version missing; not uploading ci artifacts.'
+            res = 1
+        else:
+            try:
+                check(
+                    'bazel', 'run', '//:push-build', '--',
+                    'gs://%s/%s' % (gcs, version)
```

Comment:
Will you please move the `gs://` part to the flag?

Explicitness is more important than brevity.